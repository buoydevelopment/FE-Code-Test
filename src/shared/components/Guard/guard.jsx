import React from 'react';
import { Route } from 'react-router-dom';

const Guard = (props) => {
  const { component: Component, path, isAuthenticated } = props;

  return (
    <Route exact path={path} render={() => {
      if (!isAuthenticated) {
        //Apply some auth logic
        return <div>Not authenticated</div>
      }
      return <Component />
    }} />
  );
}

export default Guard;