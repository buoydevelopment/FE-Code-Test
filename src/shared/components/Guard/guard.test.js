import React from 'react';
import { shallow } from 'enzyme';
import Guard from './guard';

const mockComponent = (
  <span>
    Random Content
  </span>
);

describe('<Guard />', () => {

  it('Should render the component when authenticated', () => {
    const wrapper = shallow(<Guard path="/"
      component={mockComponent}
      isAuthenticated={true} />);
    expect(wrapper).toMatchSnapshot();
  });

  it('Should NOT render the component when NOT authenticated', () => {
    const wrapper = shallow(<Guard path="/"
      component={mockComponent}
      isAuthenticated={false} />);
    expect(wrapper).toMatchSnapshot();
  });

});