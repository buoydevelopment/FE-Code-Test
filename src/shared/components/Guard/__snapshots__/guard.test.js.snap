// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Guard /> Should NOT render the component when NOT authenticated 1`] = `ShallowWrapper {}`;

exports[`<Guard /> Should render the component when authenticated 1`] = `ShallowWrapper {}`;
